### PR TITLE
Basic login - Finish chapter 8

### PR DESCRIPTION
**Pull request by tamdn96**

### CODE REVIEW

**Comment by tamdn96** [comment]
image

**Comment by tamdn96** [comment]
ready

File: app/views/sessions/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<% provide:title, t(".title") %>
+<h1><%= t".title"%></h1>
+
+<div class="row">
+  <div class="col-md-6 col-md-offset-3">
+    <%= form_for(:session, url: login_path) do |f| %>
+
+      <%= f.label :email %>
+      <%= f.email_field :email, class: 'form-control' %>
```
Comment (dothidiemthao) [inline app/views/sessions/new.html.erb]: dùng nháy "

File: app/views/sessions/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<% provide:title, t(".title") %>
+<h1><%= t".title"%></h1>
+
+<div class="row">
+  <div class="col-md-6 col-md-offset-3">
+    <%= form_for(:session, url: login_path) do |f| %>
```
Comment (dothidiemthao) [inline app/views/sessions/new.html.erb]: bỏ ( )

File: test/test_helper.rb [hunk 3]
```diff
@@ -1,9 +1,12 @@
 require File.expand_path("../../config/environment", __FILE__)
 require "rails/test_help"
-
+ENV[RAILS_ENV] ||= "test"
 class ActiveSupport::TestCase
   # Setup all fixtures in test/fixtures/*.yml for all tests in alphabetical order.
   fixtures :all
+  def is_logged_in?
+    !session[:user_id].nil?
```
Comment (dothidiemthao) [inline test/test_helper.rb]: dùng "session[:user_id].present?" thay cho "!session[:user_id].nil?"

File: app/helpers/sessions_helper.rb [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  def log_in user
+    session[:user_id] = user.id
+  end
+
+  def current_user
+    @current_user ||= User.find_by(id: session[:user_id])
+  end
+
+  def logged_in?
+    !current_user.nil?
```
Comment (dothidiemthao) [inline app/helpers/sessions_helper.rb]: dùng "current_user.present?" thay cho "!current_user.nil?"

**Comment by dothidiemthao** [comment]
commented

**Comment by tamdn96** [comment]
ready

**Review by dothidiemthao — APPROVED** [review]

**Comment by dothidiemthao** [comment]
merged

**Pull request merged**